### ISSUE TITLE
[기능 개선 제안] Issue Inprogress 탭 추가

### ISSUE THREAD

**Issue by E-nuri** [yona-projects/yona#291]
## 현재
이슈를 나타내는 상태가 open, closed 밖에없다.
Label을 붙여서 구분할수도 있지만 Label 을 붙여서 관리하고 있으므로 한눈에 모아보기가 어렵다.
Open | Inprogress | Closed 로 상태를 나눠서 구분할 수 있다면 

## 기대효과
팀원이 여러 이슈들 중 지금 진행중인 이슈중인지 바로 파악할 수 있게된다.
이에 따라 팀원간에 현재 진행중인 작업을 쉽게 파악할 수 있다.

![2017-09-09 4 34 12](https://user-images.githubusercontent.com/20082776/30238239-3b122568-957e-11e7-964a-f8c4276bf1f0.png)

**Comment by doortts** [comment]
@E-nuri 
흥미로운 제안 고맙습니다.
아이디어가 조금 더 필요한데요 그럼 Inprogress 는 따로 버튼으로 만들고 담당자가 버튼으로 선택하는걸로 할까요?

**Comment by E-nuri** [comment]
@doortts 
네 비슷합니다. 현재 이슈를 발행 후 이슈를 보면 `Close issue` 와 `Add a comment` 버튼이 나타납니다.

이 버튼을 살짝 바꿔서 `Inprogress` 와 `Add comment` 버튼으로 바꿔주면 될 것 같습니다.
해당 이슈 작업을 시작할 때 Inprogress 버튼을 누르면 이슈가 Inprogress 상태로 넘어가고 
작업이 끝낸뒤 이슈를 보면(Inprogress상태인 이슈를 보면) `Close issue` 버튼과 `Add a comment`가 남게되는거죠.

그럼 작업자가 작업을 끝냈으니 Close issue(또는 Comment & Close issue)버튼을 눌러서 이슈를 닫게됩니다.

**Comment by sjstyle** [comment]
개인적으로는 이 의견은 바로 반영하기 보다는 좀 더 검토가 필요할 것 같습니다.

`in progress`와 같은 라벨들을 이용해서 진행중인 것을 표시할 수 있기 때문이고, '열림'과 '닫힘' 상태가 두 개 뿐일 때는 상태 전환을 버튼 한개로 토글할 수 있지만, 상태가 3개가 될 경우에는 이런 방법으로는 상태 3개를 전환하기 어려워 집니다.

'진행중' 상태가 '열림' 상태와 완전히 관계가 없는지도 확인해봐야할 것 같습니다. 이슈가 '열려있는데 진행중' 인것인지 '진행중'이라는 별개의 상태인것인지도 애매한 것 같습니다.

**Comment by doortts** [comment]
@sjstyle 
네 만약 이슈가 제 3상태인 '진행중'을 가지게 된다면 시스템 전반에 영향을 줄 수 있는 큰 변화가 될거라 여러가지로 따져봐야 할 것 같아요. 지금도 상태는 라벨로 표시해서 쓰고 있는데 큰 불편없이 사용하고 있긴합니다.

실제로 담당자가 작업 중이라면 `조사중(uner investigation)` 이나 `작업중(working in progress)` 같은 라벨을 붙여서 이용하고 있습니다.

만약 구현하게 된다면 workflow 라는 프로젝트 옵션기능으로 제공할 수 있게 구현하면 좋을것 같습니다.

workflow 기능이란 프로젝트별로 이슈의 lifecycle을 지정할 수 있는 기능으로 열림 닫힘과는 별개로 일종의 공정처럼 미리 프로젝트에서 지정된 단계로 이슈 진행상태 변이시키는 기능을 말합니다.

`진행전(혹은 비어있는 상태)` -> `확인중` -> `작업중` -> `테스트` -> `릴리즈 대기` -> `반영완료`

와 같은 식으로요. 물론 workflow라는 말처럼 전후로 이동하거나 일종의 분기를 타서 다른 흐름으로 가거나 혹은 `진행중` 하나만 workflow로 지정해 놓아도 되도록 구성되는식으로요.

**Comment by E-nuri** [comment]
@doortts @sjstyle 
네, 저도 회사에서 작업중인 이슈에 대해서는 따로 라벨을 붙여서 사용하고있는데, 라벨이 아니라 별도의 기능(?)으로 만들면 어떨까해서 제안한 기능입니다.

요나 전반적으로 수정되어야 할 부분이 많은 기능 추가이기에 여러 영향을 고려해봐야할 듯 합니다.
